service/history/hsm/sm.go

```diff
@@ -47,7 +47,8 @@ type StateMachine[S comparable] interface {
 
 // TransitionOutput is output produced for a single transition.
 type TransitionOutput struct {
-	Tasks []Task
+	Tasks    []Task
+	IsDelete bool
```

Comment:
Maybe put this on `PathsAndOutputs` instead?

Reply:
Thanks for the suggestion. I think it's better to keep it on `TransitionOutput` because:

1. `IsDelete` represents what happened in a specific transition, rather than a property of the path.
2. `PathAndOutputs` is about where outputs occurred, while `TransitionOutput` is about what happened in each transition.
3. A path can have multiple transitions over time, and I think we want to track when deletion occurred in that sequence.

We're using `OperationLog.DeletedPaths` to track the current deleted state of paths, while `IsDelete` in `TransitionOutput` lets us track when and where deletions happened in the transition history. This separation of concerns feels cleaner.

Reply:
Ah, this shouldn't be here because `TransitionOutput` is used by the application. `PathsAndOutputs` should only be used by the framework.

Reply:
Fixed